share/functions/__fish_print_make_targets.fish

```diff
@@ -1,13 +1,11 @@
-function __fish_print_make_targets
-    # Some seds (e.g. on Mac OS X), don't support \n in the RHS
-    # Use a literal newline instead
-    # http://sed.sourceforge.net/sedfaq4.html#s4.1
-    # The 'rev | cut | rev' trick removes everything after the last colon
-    for file in GNUmakefile Makefile makefile
+function __fish_print_make_targets --argument directory
+    if test -z "$directory"
+        set directory '.'
+    end
+
+    for file in $directory/{GNUmakefile,Makefile,makefile}
         if test -f $file
-            __fish_sgrep -h -o -E '^[^#%=$[:space:]][^#%=$]*:([^=]|$)' $file ^/dev/null | rev | cut -d ":" -f 2- | rev | sed -e 's/^ *//;s/ *$//;s/  */\\
-/g' ^/dev/null
-            # On case insensitive filesystems, Makefile and makefile are the same; stop now so we don't double-print 
+            make -C $directory -prRn | awk -v RS= -F: '/^# File/,/^# Finished Make data base/ {if ($1 !~ "^[#.]") {print $1}}' ^/dev/null
```

Comment:
That latter one (after removing the "-C source/server" part) prints some weird additional stuff in the fish repo. In particular: `echo "  CPPFLAGS = '-DLOCALEDIR=\"/usr/local/share/locale\" -DPREFIX=L\"/usr/local\" -DDATADIR=L\"/usr/local/share\" -DSYSCONFDIR=L\"/usr/local/etc\" -DBINDIR=L\"/usr/local/bin\" -DDOCDIR=L\"/usr/local/share/doc/fish\" -iquote. -iquote./src/ -DFISH_BUILD_VERSION=\"2.4.0-216-g2db0bbc\"'" ||`.